code-challenges.test.js

```diff
@@ -12,30 +14,120 @@
+
+// I'm going to try running this as a for-loop. I'm going to set fibArray = the first 12 numbers of the Fibonacci sequence and have the function iterate over the length of that array for the length of the given index length.
+
 // b) Create the function that makes the test pass.
+const fibArray = [1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144];
+// const fibonacciSequence = (num) => {
+//   const newFibArray = [];
+//   for (let i = 0; i < fibArray.length; i++) {
+//     newFibArray.push(fibArray.slice(0, num));
+//     return newFibArray;
+//   }
+// };
+// I basically have the correct output, but it's failing because the array is getting an extra set of brackets somewhere and I'm not sure where.
+
+// I figured out the issue, using slice, I was taking a subset of the array I had created and pushing that sliced array to the empty array I had declared. This was causing the test to fail because of the extra set of brackets that were contained within the output.
+
+// I was correct in that I would need to iterate over an array somehow, but I wasn't exactly sure. I understand the Fibonacci sequence quiet well, it's constant representation in the natural world really is fascinating to me. Be that as it may, it doesn't help the abscence of its representation here.
+
+// I declared a new function with a for-loop specifically for creating the fibonacci sequence equal in length to the given number.
+
+const fibonacciSequence = (num) => {
+  // created a new array set to an empty array as a place to store the pushed Fibonacci values.
+  const newFibArray = [];
+
+  // Establishing two variables that will serve as the beginning of the Fibonacci sequence. [1,1,2,...etc]
+  let n1 = 1;
+  let n2 = 1;
+
+  // This will allow the loop to iterate for less than the length of the given number and allow for the correct number of indexes as well as brackets, since I am now simply pushing values to the empty array.
+
+  for (let i = 0; i < num; i++) {
+    // fibSeq represents the beginning of the Fibonacci count that adds the previous two numbers together.
+    let fibSeq = n1 + n2;
+
+    // The values of n1 are being continuously pushed to the new array creating the Fibonacci sequence
+    newFibArray.push(n1);
+
+    // The n1 and n2 variables previously set to 1 are now updated in order to continue with the count of the Fibonacci sequence and return the correct newFibArray length.
+    n1 = n2;
+    n2 = fibSeq;
+  }
+  return newFibArray;
+};
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+// I was correct! Not only was I able to get a test pass, but with this function, the function is now able to repeat the process with any given number and return the correct Fibonacci sequence.
```

Comment:
Excellent work, and I really appreciate your walkthrough in pseudocode on your entire process!